experimental/emulated_vault/opt/emulated-vault/adapter/base.py

```diff
@@ -0,0 +1,292 @@
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#	 http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+#
+
+import os
+
+class Base(object):
```

Comment:
Wow. Did not know python has that!
It would however require some restructuring of the code:
The adapter base class would become a "db" class that gets a "storage adapter"
Storage adapter would be abstract with fs as implementation.
Makes sense?

Reply:
Yeah that sounds fine